### PR TITLE
remove platform lib/dai.h from SOF Zephyr builds

### PR DESCRIPTION
**Pull request by kv2019i**
A patch series the convert remaining use of platform lib/dai.h in SOF cmmon code to either device-tree (in generic code) or simple ifdefs (in vendor specific compiliation units).

In the end of series, remove the lib/dai.h definitions for all Zephyr-only SOF platforms.

Link: https://github.com/thesofproject/sof/issues/5794

### CODE REVIEW

**Comment by kv2019i** [comment]
Stub build is failing. I think https://github.com/thesofproject/sof/pull/9651 will help with this. Current chain_dma.c require HDA drivers to be enabled in the build. Will confirm.

File: src/include/ipc4/alh.h [hunk 1]
```diff
@@ -36,10 +36,19 @@
...
+#else
+#define IPC4_DAI_NUM_ALH_BI_DIR_LINKS	0
+#define IPC4_DAI_NUM_ALH_BI_DIR_LINKS_GROUP	0
+#endif
+
 /* copier id = (group id << 4) + codec id + IPC4_ALH_DAI_INDEX_OFFSET
  * dai_index = (group id << 8) + codec id;
  */
-#define IPC4_ALH_DAI_INDEX(x) ((((x) & 0xF0) << DAI_NUM_ALH_BI_DIR_LINKS_GROUP) + \
+#define IPC4_ALH_DAI_INDEX(x) ((((x) & 0xF0) << IPC4_DAI_NUM_ALH_BI_DIR_LINKS_GROUP) + \
```
Comment (lgirdwood) [inline src/include/ipc4/alh.h:51]: Btw, should this be in Zephyr now ?

File: src/audio/base_fw_intel.c [hunk 2]
```diff
@@ -47,6 +47,10 @@ struct ipc4_modules_info {
 #define PLATFORM_LPSRAM_EBB_COUNT	(DT_REG_SIZE(DT_NODELABEL(sram1)) / SRAM_BANK_SIZE)
 #define PLATFORM_HPSRAM_EBB_COUNT	(DT_REG_SIZE(DT_NODELABEL(sram0)) / SRAM_BANK_SIZE)
 
+#define DT_NUM_SSP_BASE		DT_NUM_INST_STATUS_OKAY(intel_ssp)
```
Comment (lgirdwood) [inline src/audio/base_fw_intel.c:51]: Whats happens if status not okay in DT ? Do we need a DT_xxxxx_OR() usage here to give a meaningful value if SSP is not enabled ?

File: src/audio/base_fw_intel.c [hunk 3]
```diff
@@ -88,7 +89,7 @@ int basefw_vendor_hw_config(uint32_t *data_offset, char *data)
 	tuple = tlv_next(tuple);
 	/* 2 DMIC dais */
 	value =  DT_NUM_SSP_BASE + DT_NUM_HDA_IN + DT_NUM_HDA_OUT +
-			DAI_NUM_ALH_BI_DIR_LINKS + 2;
+			IPC4_DAI_NUM_ALH_BI_DIR_LINKS + 2;
```
Comment (lgirdwood) [inline src/audio/base_fw_intel.c:92]: DT data in the future ?

**Review by lgirdwood — COMMENTED** [review]
Good stuff - nice to see more deletions of legacy code.

File: src/include/ipc4/alh.h (hunk 1 above)
Comment (kv2019i) [reply src/include/ipc4/alh.h:51]: @lgirdwood I took a look at this, but we don't really have a good schema for ALH instances in DT, and on newer platforms (ace20 onwards), this is really replaced by the HDA link DMA. And to add to that, this bi-dir-link group is really only used to parse the blob, so it's not even strictly tied to hw interface and its use, so in the end did not seem worth the effort.

File: src/audio/base_fw_intel.c (hunk 2 above)
Comment (kv2019i) [reply src/audio/base_fw_intel.c:51]: This the NUM variant, so it will return 0 if no matches found.

File: src/audio/base_fw_intel.c (hunk 3 above)
Comment (kv2019i) [reply src/audio/base_fw_intel.c:92]: See comment one, it seems this bi-dir-link property is a legacy one.

**Comment by kv2019i** [comment]
V2 pushed:
 - rebased to latest main to take https://github.com/thesofproject/sof/pull/9651 in 
 - at least locally the fuzzer builds are now passing, let's see what CI says

**Review by lyakh — APPROVED** [review]

**Review by tmleman — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**